src/axolotl/utils/data.py

```diff
@@ -102,13 +102,22 @@ def load_tokenized_prepared_datasets(
                 pass
 
             # prefer local dataset, even if hub exists
-            if Path(d.path).exists():
-                ds = load_dataset(
-                    "json",
-                    data_files=d.path,
-                    streaming=False,
-                    split=None,
-                )
+            local_path = Path(d.path)
+            if local_path.exists():
+                if local_path.is_dir():
+                    ds = load_dataset(
+                        d.path,
+                        data_files=d.data_files,
+                        streaming=False,
+                        split=None,
+                    )
+                else:  # if local_path.is_file():
```

Comment:
Perhaps change to elif?

Reply:
I was a bit hesitated, because I'm used to writing `if...elif...else` but there seems to be nothing to write for `else`

Reply:
Would it ever reach an else afterward? Do we need to raise Error?

Reply:
OK